pkg/providers/aws/provider_postgres.go

```diff
@@ -700,14 +700,19 @@ func buildRDSUpdateStrategy(rdsConfig *rds.CreateDBInstanceInput, foundConfig *r
 			return nil, errorUtil.Wrap(err, "invalid postgres version")
 		}
 		if engineUpgradeNeeded {
+			logrus.Info(fmt.Sprintf("Engine upgrade found, the current EngineVersion is %s and is upgrading to %s", *foundConfig.EngineVersion, *rdsConfig.EngineVersion))
 			mi.EngineVersion = rdsConfig.EngineVersion
 			if cr.Spec.ApplyImmediately {
 				mi.ApplyImmediately = aws.Bool(cr.Spec.ApplyImmediately)
 			}
 			updateFound = true
 		}
 	}
-	if !updateFound || !verifyPendingModification(mi, foundConfig.PendingModifiedValues) {
+	if mi.ApplyImmediately != nil {
+		if (!updateFound || !verifyPendingModification(mi, foundConfig.PendingModifiedValues)) && !*mi.ApplyImmediately {
+			return nil, nil
+		}
+	} else if !updateFound || !verifyPendingModification(mi, foundConfig.PendingModifiedValues) {
```

Comment:
code suggestion but the logic is the same either way.
```suggestion
		if (!updateFound || !verifyPendingModification(mi, foundConfig.PendingModifiedValues)) && (mi.ApplyImmediately == nil || !*mi.ApplyImmediately) {
```

Reply:
this looks a bit tidier :+1: